### DISCUSSION TITLE
Why does only the first p turn red?

### DISCUSSION THREAD

**Discussion by 332lfgb6** [tailwindlabs/tailwindcss#12602] [Help] [answered]
```vue
<template>
  <div class="group">
    <p class="group-[:nth-of-type(1)_&]:bg-red-500">123</p>
    <p class="group-[:nth-of-type(2)_&]:bg-red-500">123</p>
    <p class="group-[:nth-of-type(3)_&]:bg-red-500">123</p>
  </div>
</template>
```
https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state
![image](https://github.com/tailwindlabs/tailwindcss/assets/51245335/db48557a-0ae1-428e-a7c7-f50b61c4d8bc)
![image](https://github.com/tailwindlabs/tailwindcss/assets/51245335/cef6e673-5860-44c8-853e-c91da92587c9)

**Comment by wongjn** [comment]
Assuming this is from a Vue component, I would posit that an ancestor matches the `:nth-of-type(1)` portion of the selector construct but does no ancestor matches the `:nth-of-type(2)` or `:nth-of-type(3)` portion of the other selectors.

**Reply by wongjn** [reply]
Replying to https://github.com/tailwindlabs/tailwindcss/discussions/12602#discussioncomment-7870091:

> @wongjn
> 
> hello, How do I make the second p red?
> 
> ```
> <div class="group">
>     <p class="first:bg-red-500">123</p>
>     <p class="[:nth-child(2)]:bg-red-500">123</p>
>     <p class="last:bg-red-500">123</p>
>   </div>
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/51245335/290984145-bfb32bb6-ea18-4896-8725-77b08fe78a06.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI3MjAyMTMsIm5iZiI6MTcwMjcxOTkxMywicGF0aCI6Ii81MTI0NTMzNS8yOTA5ODQxNDUtYmZiMzJiYjYtZWExOC00ODk2LTg3MjUtNzdiMDhmZTc4YTA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE2VDA5NDUxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwNjJjZWYzNzExZDQ1NjlhNjQ5MTE1ZTVhNDJlZTM2NzVmOTY5MDc1YzVjNzNhYTk4MTM2NzA2ZmNiZWY4OTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.h-tRwNXqZSF1X89d61bjMQH5YwurqEewRrSYkOriDpY)

You can use `bg-red-500` without any variant:
```diff
 <div class="group">
   <p class="first:bg-red-500">123</p>
-  <p class="[:nth-child(2)]:bg-red-500">123</p>
+  <p class="bg-red-500">123</p>
   <p class="last:bg-red-500">123</p>
 </div>
```
Or to have the seemingly unneccessary `:nth-child(2)` qualifier apply:

```diff
 <div class="group">
   <p class="first:bg-red-500">123</p>
-  <p class="[:nth-child(2)]:bg-red-500">123</p>
+  <p class="[&:nth-child(2)]:bg-red-500">123</p>
   <p class="last:bg-red-500">123</p>
 </div>
```

**Reply by 332lfgb6** [reply]
I didn't mean it that way, but I solved my problem, which is to write it in style, tks

**Reply by wongjn** [reply]
Could you explain more how you meant it please?

**Reply by 332lfgb6** [reply, marked as answer]
I thought class="group-[:nth-of-type(1)_ &]:bg-red-500" means: Make the background color of the first p label red until I see that you wrote"class="[&:nth-child(2)]:bg-red-500", I realized that I had misunderstood the meaning of class="group-[:nth-of-type(1)_&]:bg-red-500", which actually refers to the first of the outer div tags. thank you for your help.😄

**Comment by 332lfgb6** [comment]
@wongjn 

hello, How do I make the second p red?
```vue
<div class="group">
    <p class="first:bg-red-500">123</p>
    <p class="[:nth-child(2)]:bg-red-500">123</p>
    <p class="last:bg-red-500">123</p>
  </div>
```
![image](https://github.com/tailwindlabs/tailwindcss/assets/51245335/bfb32bb6-ea18-4896-8725-77b08fe78a06)

**Reply by wongjn** [reply]
Replied at https://github.com/tailwindlabs/tailwindcss/discussions/12602#discussioncomment-7870134 to keep the reply chain together.
